Add ChaosContext and ChaosTargetFilter in ChaosParameters and tests in SFCTL

motanv opened the pull request:
Fixes # .

Changes include:

-
-
-

Verifed:

- [ ] Build and test CI passes
- [ ] History and readme updated to reflect changes
- [ ] Package version updated according to semantic versioning rules
- [ ] Tests modified or added, when applicable

samedder commented on src/sfctl/custom_chaos.py:
```diff
@@ -43,6 +72,60 @@ def start( #pylint: disable=too-many-arguments,too-many-locals
     entry specifies as a key the application type name and as  a value an
     integer that represents the MaxPercentUnhealthyApplications percentage
     used to evaluate the applications of the specified application type.
+    :param str context: JSON encoded map of (string, string) type key-value
```
Can you instead move this help string into a new file in `src/sfctl/helps`. That's where all new docstrings should go since it supports more features and YAML formatting.

motanv replied:
Moved

Christina-Kang replied:
I think it should still work when you remove it from here... can you confirm? Thanks! :)

samedder commented on src/sfctl/tests/chaos_test.py:
```diff
@@ -0,0 +1,66 @@
+import unittest
+import sfctl.custom_chaos as sf_c
+
+#pylint: disable=invalid-name
```
Why do you need to disable this?

motanv replied:
Copy-paste from health_test.py, removed now.

samedder commented:
Oh, please also update the PR title

Christina-Kang commented on src/sfctl/custom_chaos.py:
```diff
@@ -4,16 +4,45 @@
+def parse_chaos_context(formatted_chaos_context):
+    """"Parse a chaos context from a formatted context"""
+    from azure.servicefabric.models.chaos_context import (
+    ChaosContext
```
Indenting seems wrong here

samedder commented:
@Christina-Kang @motanv I presume this is also pending the 6.1 SDK release?

Christina-Kang commented:
@samedder Yes, this is pending 6.1

motanv commented:
@samedder @Christina-Kang Working on it, I shall complete this by tomorrow, need to talk to likevi as he has some work on CLI as well.

motanv commented:
@samedder What do you mean by History/Version info? When you said the following?

"Please also make sure the following are updated:

CodeOwners
History/Version info
And make sure PyLint passes too."

motanv commented:
@samedder @Christina-Kang I think I have addressed all comments and the CI errors except the following: 
======================================================================
ERROR: Parse application inclusion list
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/Azure/service-fabric-cli/src/sfctl/tests/chaos_test.py", line 62, in parse_application_list_test
    'ApplicationInclusionList': ['fabric:/TestApp1', 'fabric:/TestApp2'] #pylint: disable=line-too-long
  File "/home/travis/build/Azure/service-fabric-cli/src/sfctl/custom_chaos.py", line 22, in parse_chaos_target_filter
    from azure.servicefabric.models.chaos_target_filter import (
ImportError: No module named chaos_target_filter
======================================================================
ERROR: Parse nodetypeinclusionlist list
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/Azure/service-fabric-cli/src/sfctl/tests/chaos_test.py", line 48, in parse_nodetype_list_test
    'N0010Ref', 'N0020Ref', 'N0030Ref', 'N0070Ref']
  File "/home/travis/build/Azure/service-fabric-cli/src/sfctl/custom_chaos.py", line 22, in parse_chaos_target_filter
    from azure.servicefabric.models.chaos_target_filter import (
ImportError: No module named chaos_target_filter
======================================================================
ERROR: Parse None chaos target filter returns None
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/Azure/service-fabric-cli/src/sfctl/tests/chaos_test.py", line 41, in parse_none_target_filter_test
    self.assertIs(sf_c.parse_chaos_target_filter(None), None)
  File "/home/travis/build/Azure/service-fabric-cli/src/sfctl/custom_chaos.py", line 22, in parse_chaos_target_filter
    from azure.servicefabric.models.chaos_target_filter import (
ImportError: No module named chaos_target_filter

I think the CI is validating against an older model (not release_6.1) that is why it cannot find chaos_target_filter.py

Could you check why that is the case?

Christina-Kang commented on src/README.rst:
```diff
@@ -16,11 +16,14 @@ To get started, after installation run the following:
 Change Log
 ==========
 
+3.2.0
```
Left over line?

motanv replied:
Removed

Christina-Kang commented on src/README.rst:
```diff
@@ -16,11 +16,14 @@ To get started, after installation run the following:
 
 - Update to official 6.0.2 Service Fabric SDK
 - Property command group added
+- ChaosContext (context) and ChaosTargetFilter (chaos-target-filter) arguments are added to Chaos start command (#62)
```
Forgot to ask last time, are the chaos changes breaking changes? If so, we should make this version 4.0.0. If not, then this is good. Thanks!

motanv replied:
This is not breaking change.

Christina-Kang commented on src/sfctl/helps/chaos.py:
```diff
@@ -0,0 +1,184 @@
+
```
Remove white line. Thanks!

Christina-Kang commented on src/sfctl/tests/chaos_test.py:
```diff
@@ -0,0 +1,68 @@
+# license information.
+# -----------------------------------------------------------------------------
+
+"""Custom Chaos command related tests"""
```
Add newline here.

Christina-Kang commented on src/sfctl/tests/chaos_test.py:
```diff
@@ -0,0 +1,68 @@
+
+    def parse_nodetype_list_test(self):
+        """Parse nodetypeinclusionlist list"""
+
```
Don't need this newline here.

tanv1r replied:
It appears in other tests, e.g., health_test, a newline is introduced after function def and import (no newline between def and import but after that there is a newline). I thought my newline belongs to that type of newline as I do not have an import.

motanv replied:
tanv1r is actually me, I was signed in with a different id

Christina-Kang commented:
@motanv There are a few lint errors shown through Travis CI. Could you fix those please? Other than that, looks good to me. Thanks!

samedder commented:
@motanv you won't be able to complete this PR until we publish the 6.1 SDK so I'll ping this thread when we update to the 6.1 SDK.

samedder commented:
@motanv thanks for the contribution! 🎉
